torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2706,6 +2707,24 @@ options :class:`~torch.distributed.ProcessGroupNCCL.Options`).
               },
               py::arg("timeout"),
               py::call_guard<py::gil_scoped_release>())
+          .def(
+              "_set_ephemeral_timeout",
+              [](const c10::intrusive_ptr<::c10d::ProcessGroupNCCL>& self,
+                 const std::chrono::milliseconds& timeout) {
+                self->setEphemeralTimeout(timeout);
+              },
+              py::arg("timeout"),
+              py::call_guard<py::gil_scoped_release>())
+          .def(
+              "_check_work_timeout",
+              [](const c10::intrusive_ptr<::c10d::ProcessGroupNCCL>& self,
+                 const c10::intrusive_ptr<::c10d::Work> work,
+                 const std::chrono::milliseconds& timeout) {
+                return self->checkWorkTimeout(work, timeout);
+              },
+              py::arg("work"),
+              py::arg("timeout"),
+              py::call_guard<py::gil_scoped_release>())
```

Comment:
just fyi: these are very cheap functions, if you weren't taking locks inside them it wouldn't be necessary to release GIL on them